lib/python/qmk/cli/new/keyboard.py

```diff
@@ -66,18 +66,19 @@ def new_keyboard(cli):
     keyboard_type = cli.args.type if cli.args.type else choice('Keyboard Type:', KEYBOARD_TYPES, default=0)
 
     # Get username
-    user_name = None
-    while not user_name:
-        user_name = question('Your GitHub User Name:', default=find_user_name())
+    user_name = cli.args.username
+    if user_name:
+        cli.log.info(f'Using GitHub User Name: {user_name}')
+    else:
+        user_name = question('Your GitHub User Name:', default=cli.config.user.name if cli.config.user.name else git_get_username())
 
         if not user_name:
             cli.log.error('You didn\'t provide a username, and we couldn\'t find one set in your QMK or Git configs. Please try again.')
+            return False
```

Comment:
```suggestion

```
As `question` will keep asking until something is provided, I don't think we will ever hit this code.

Reply:
Agreed. But that is a separate refactoring, not directly related to what the changeset is doing: muting the question.

Reply:
Actually ... `question` will accept an empty answer if the default is empty. That is what this check is catching. But I happened to change so it no longer keeps asking in that case. I will fix that.